src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // write your code here
+const promise1 = new Promise((resolve) => {
+  document.querySelector('.logo').addEventListener('click', () => {
```

Comment:
The querySelector method is not the most efficient way to select elements by their class name. Consider using the getElementsByClassName or querySelectorAll methods instead. These methods return a live HTMLCollection of elements, which can be more efficient when dealing with large numbers of elements.